components/validator/resources.rst

```diff
@@ -0,0 +1,200 @@
+class itself. The ``Validator`` needs to be able to retrieve this metadata
+from the file or class. To do that, it uses a set of loaders.
+
+.. seealso::
+
+    You'll learn how to define the metadata in :doc:`metadata`.
+
+The StaticMethodLoader
+----------------------
+
+The most basic loader is the
+:class:`Symfony\\Component\\Validator\\Mapping\\Loader\\StaticMethodLoader`.
+This loader will call a static method of the class in order to get the
+metadata for that class. The name of the method is configured using the
+:method:`Symfony\\Component\\Validator\\ValidatorBuilder::addMethodMapping`
+method of the Validator builder::
+
+    use Symfony\Component\Validator\Validation;
+
+    $validator = Validation::createValidatorBuilder()
+        ->addMethodMapping('loadValidatorMetadata')
+        ->getValidator();
+
+Now, the retrieved ``Validator`` tries to find the ``loadValidatorMetadata()``
+method of the class to validate to load its metadata::
+
+    use Symfony\Component\Validator\Mapping\ClassMetadata;
+    use Symfony\Component\Validator\Constraints\NotBlank;
+    use Symfony\Component\Validator\Constraints\Length;
+
+    class User
+    {
+        protected $name;
+
+        public static function loadValidatorMatadata(ClassMetadata $metadata)
+        {
+            $metadata->addPropertyConstraint('name', new NotBlank());
+            $metadata->addPropertyConstraint('name', new Length(array(
+                'min' => 5,
+                'max' => 20,
+            )));
+        }
+    }
+
+.. tip::
+
+    You can call this method multiple times to add multiple supported method
+    names. You can also use
+    :method:`Symfony\\Component\\Validator\\ValidatorBuilder::addMethodMappings`
+    to set an array of supported method names.
+
+The FileLoaders
+---------------
+
+The component also provides 2 file loaders, one to load Yaml files and one to
+load XML files. Use 
+:method:`Symfony\\Component\\Validator\\ValidatorBuilder::addYamlMapping` or
+:method:`Symfony\\Component\\Validator\\ValidatorBuilder::addXmlMapping` to
+configure the locations of these files::
+
+    use Symfony\Component\Validator\Validation;
+
+    $validator = Validation::createValidatorBuilder()
+        ->addYamlMapping('config/validation.yml')
+        ->getValidator();
+
+.. tip::
+
+    Just like with the method mappings, you can also use 
+    :method:`Symfony\\Component\\Validator\\ValidatorBuilder::addYamlMappings` and
+    :method:`Symfony\\Component\\Validator\\ValidatorBuilder::addXmlMappings`
+    to configure an array of file paths.
+
+The AnnotationLoader
+--------------------
+
+At last, the component provides an
+:class:`Symfony\\Component\\Validator\\Mapping\\Loader\\AnnotationLoader`.
+This loader uses an annotation reader to parse the annotations of a class.
+Annotations are placed in doc block comments (`/** ... */`) and start with an
+``@``. For instance::
+
+    // ...
+
+    /**
+     * @Assert\NotBlank()
```

Comment:
please also add the use statement before `// ...` in this code block: `use Symfony\Component\Validator\Constraints as Assert;`